src/browser/BrowserCodeReader.ts

```diff
@@ -699,17 +699,15 @@ export class BrowserCodeReader {
         setTimeout(() => loop(), this.timeBetweenScansMillis);
       } catch (e) {
 
+        callbackFn(null, e);
+
         const isChecksumOrFormatError = e instanceof ChecksumException || e instanceof FormatException;
         const isNotFound = e instanceof NotFoundException;
 
-        if (!isChecksumOrFormatError && !isNotFound) {
-          // not expected
-          throw e;
+        if (isChecksumOrFormatError || isNotFound) {
+          // trying again
+          setTimeout(() => loop(), this.timeBetweenScansMillis);
```

Comment:
I changed the timeout from 0  to `this.timeBetweenScansMillis` as the previous version drastically increased the workload of the thread without any chance of configuration.

Any concerns?

Reply:
Yep, that time between scans was meant to be used between successful scans, not every loop, because the idea here is to try as much as it can so it can decode as fast as possible and, when successful scans something, it waits a little before scan again so it avoids decoding the same code multiple not-needed times.

We could create a new property to configure a delay between decode attempts, so you could reduce the thread size and not mix behaviors.

Reply:
Reverted the change and may open a new merge request these days to introduce the new configuration property.